Updating a client protocol mappers through Admi CLI (kcadm) resets the client service account roles

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/cli

### Describe the bug

I tried to update the protocol mappers of a client using the Admin CLI like this : 

`kcadm update clients/<client_id> -f client-protocol-mappers.json`

With the json file containing the id of the client and the updated protocol mappers. Here is a simple example : 
```
{
  "id" : "1c8c3cb0-4496-4ef3-bf14-93a40c67dc6f",
  "protocolMappers" : [ {
    "id" : "25c9005f-4348-4a13-8c39-4dd5bb3cb30d",
    "name" : "given name",
    "protocol" : "openid-connect",
    "protocolMapper" : "oidc-usermodel-property-mapper",
    "consentRequired" : false,
    "config" : {
      "userinfo.token.claim" : "true",
      "user.attribute" : "firstName",
      "id.token.claim" : "false",
      "access.token.claim" : "true",
      "claim.name" : "given_name",
      "jsonType.label" : "String"
    }
  }, {
    "id" : "d3a75456-fc69-432f-9a80-5825703ff84b",
    "name" : "family name",
    "protocol" : "openid-connect",
    "protocolMapper" : "oidc-usermodel-property-mapper",
    "consentRequired" : false,
    "config" : {
      "userinfo.token.claim" : "true",
      "user.attribute" : "lastName",
      "id.token.claim" : "false",
      "access.token.claim" : "true",
      "claim.name" : "family_name",
      "jsonType.label" : "String"
    }
  } ]
}
```

Everything works fine and the protocol mappers are correctly updated for my client.
All other client configuration values still have correctly the same values as before the Admin CLI command was executed (Client name, description, redirect URIs, theme, keys, credentials, etc...), except the configuration of the service account roles which got reset to the default values.

Before : 
![image](https://github.com/keycloak/keycloak/assets/11772429/e8f16c1d-0564-40ac-866b-b0fc6ecb9eca)

After updating the protocol mappers : 
![image](https://github.com/keycloak/keycloak/assets/11772429/b0145bc4-fc80-4040-a0f6-95c191931f94)

As you can see no more roles (except the default 'default-roles-<realm>' role) are associated to the service account.






### Version

24.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

Only the protocol mappers of the client should be updated

### Actual behavior

Protocol mappers of the client are correctly updated but the service account roles are reset

### How to Reproduce?

Create a client in the management console.
Associate roles to the service account of the client (through the 'Service account roles' tab)
Execute kcadm update clients/<client_id> -f client-protocol-mappers.json
Refresh the page showing the roles associated to the service account of the client -> Roles are no more associated.

### Anything else?

_No response_

> @omasseau can you provide a full stacktrace? Check the server log, and/or include -x with the kcadm command.

Here is the generated stacktrace when adding the -x option : 

`kcadm update --merge clients/<client_id> -f client-protocol-mappers.json -x`

```
null [RESTEASY003065: Cannot consume content type]
org.keycloak.client.admin.cli.util.HttpResponseException: null [RESTEASY003065: Cannot consume content type]
        at org.keycloak.client.admin.cli.util.HeadersBodyStatus.checkSuccess(HeadersBodyStatus.java:61)
        at org.keycloak.client.admin.cli.util.HttpUtil.checkSuccess(HttpUtil.java:353)
        at org.keycloak.client.admin.cli.commands.AbstractRequestCmd.process(AbstractRequestCmd.java:363)
        at org.keycloak.client.admin.cli.commands.AbstractRequestCmd.execute(AbstractRequestCmd.java:126)
        at org.jboss.aesh.console.command.container.DefaultCommandContainer.executeCommand(DefaultCommandContainer.java:63)
        at org.jboss.aesh.console.command.container.DefaultCommandContainer.executeCommand(DefaultCommandContainer.java:48)
        at org.keycloak.client.admin.cli.aesh.AeshConsoleCallbackImpl.execute(AeshConsoleCallbackImpl.java:54)
        at org.jboss.aesh.console.AeshProcess.run(AeshProcess.java:53)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.RuntimeException: {"error":"RESTEASY003065: Cannot consume content type"}
        ... 11 more
```

--- a/integration/client-cli/admin-cli/src/main/java/org/keycloak/client/admin/cli/commands/AbstractRequestCmd.java
+++ b/integration/client-cli/admin-cli/src/main/java/org/keycloak/client/admin/cli/commands/AbstractRequestCmd.java
@@ -29,6 +29,7 @@ import org.keycloak.client.cli.util.HeadersBodyStatus;
 import org.keycloak.client.cli.util.HttpUtil;
 import org.keycloak.client.cli.util.OutputFormat;
 import org.keycloak.client.cli.util.ReturnFields;
+import org.keycloak.util.JsonSerialization;
 
 import java.io.BufferedInputStream;
 import java.io.ByteArrayInputStream;
@@ -265,6 +266,11 @@ public abstract class AbstractRequestCmd extends AbstractAuthOptionsCmd {
                 ReflectionUtil.merge(ctx.getResult(), (ObjectNode) ctxremote.getResult());
             }
             ctx = ctxremote;
+            try {
+                ctx.setContent(JsonSerialization.writeValueAsString(ctxremote.getResult()));
+            } catch (IOException e) {
+                throw new RuntimeException("Could not convert merge result to string " + e.getMessage(), e);
+            }
         }
 
         if (attrs.size() > 0) {

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/cli/admin/KcAdmUpdateTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/cli/admin/KcAdmUpdateTest.java
@@ -162,17 +162,28 @@ public class KcAdmUpdateTest extends AbstractAdmCliTest {
             // test using merge with file
             exe = KcAdmExec.newBuilder()
                     .argsLine("update clients/" + client.getId() + " --config '" + configFile.getName() +
-                            "' -o -s enabled=true -m -f -")
+                            "' -o -m -f -")
                     .stdin(new ByteArrayInputStream("{ \"webOrigins\": [\"http://localhost:8980/myapp\"] }".getBytes()))
                     .execute();
 
             assertExitCodeAndStdErrSize(exe, 0, 0);
 
-
             client = JsonSerialization.readValue(exe.stdout(), ClientRepresentation.class);
             Assert.assertEquals("webOrigins", Arrays.asList("http://localhost:8980/myapp"), client.getWebOrigins());
-            Assert.assertTrue("enabled is true", client.isEnabled());
+            Assert.assertFalse("enabled is false", client.isEnabled());
             Assert.assertEquals("redirectUris", Arrays.asList("http://localhost:8980/myapp/*"), client.getRedirectUris());
+
+            exe = KcAdmExec.newBuilder()
+                    .argsLine("update clients/" + client.getId() + " --config '" + configFile.getName() +
+                            "' -o -s enabled=true -m -f -")
+                    .stdin(new ByteArrayInputStream("{ \"webOrigins\": [\"http://localhost:8980/myapp1\"] }".getBytes()))
+                    .execute();
+
+            assertExitCodeAndStdErrSize(exe, 0, 0);
+
+            client = JsonSerialization.readValue(exe.stdout(), ClientRepresentation.class);
+            Assert.assertEquals("webOrigins", Arrays.asList("http://localhost:8980/myapp1"), client.getWebOrigins());
+            Assert.assertTrue("enabled is true", client.isEnabled());
         }
     }
 }

